gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStore.java

```diff
@@ -180,9 +181,7 @@ public Set<String> getFailedDagIds() throws IOException {
   @Override
   // todo - updating different maps here and in addDagNodeState can result in inconsistency between the maps
   public synchronized void deleteDagNodeState(DagManager.DagId dagId, Dag.DagNode<JobExecutionPlan> dagNode) {
-    this.jobToDag.remove(dagNode);
```

Comment:
these should be replaced with calls to the dagActionStore right to delete/add dag related stuff? otherwise why do we have these removals

Reply:
actually these fields were unused. that's the objective 2 for this PR